dissect/formats/mbr.py

```diff
@@ -0,0 +1,119 @@
+'''
+Structures useful for parsing an MS-DOS style MBR.
+'''
+import os
+import math
+import logging
+import contextlib
+import collections
+
+import vstruct.types as v_types
+
+
+# assumption: sectors are 512-bytes in lenght
+# anything other than this is... non-standard, and weird.
+SECTOR_SIZE = 512
+
+
+# partition types
+SYSTEMID = v_types.venum()
+SYSTEMID.EMPTY           = 0
+SYSTEMID.FAT_12       = 1
+SYSTEMID.XENIX_ROOT      = 2
+SYSTEMID.XENIX_USR       = 3
+SYSTEMID.FAT_16_INF32MB  = 4
+SYSTEMID.EXTENDED        = 5
+SYSTEMID.FAT_16          = 6
+SYSTEMID.NTFS_HPFS       = 7
+SYSTEMID.AIX             = 8
+SYSTEMID.AIX_BOOT        = 9
+SYSTEMID.OS2_BOOT_MGR    = 10
+SYSTEMID.PRI_FAT32_INT13 = 11
+SYSTEMID.EXT_FAT32_INT13 = 12
+SYSTEMID.EXT_FAT16_INT13 = 14
+SYSTEMID.WIN95_EXT       = 15
+SYSTEMID.OPUS            = 16
+SYSTEMID.FAT_12_HIDDEN   = 17
+SYSTEMID.COMPAQ_DIAG     = 18
+SYSTEMID.FAT_16_HIDDEN_INF32MB = 20
+SYSTEMID.FAT_16_HIDDEN   = 22
+SYSTEMID.NTFS_HPFS_HIDDEN= 23
+SYSTEMID.VENIX           = 64
+SYSTEMID.NOVEL0          = 81
+SYSTEMID.MICROPORT       = 82
+SYSTEMID.GNU_HURD        = 99
+SYSTEMID.NOVEL1          = 100
+SYSTEMID.PC_IX           = 117
+SYSTEMID.MINUX_OLD       = 128
+SYSTEMID.MINUX_LINUX     = 129
+SYSTEMID.LINUX_SWAP      = 130
+SYSTEMID.LINUX_NATIVE    = 131
+SYSTEMID.AMOEBA          = 147
+SYSTEMID.AMOEBA_BBT      = 148
+SYSTEMID.BSD_386         = 165
+SYSTEMID.BSDI_FS         = 183
+SYSTEMID.BSDI_SWAP       = 184
+SYSTEMID.SYRINX          = 199
+SYSTEMID.CP_M            = 219
+SYSTEMID.ACCESS_DOS      = 225
+SYSTEMID.DOS_R_O         = 227
+SYSTEMID.DOS_SECONDARY   = 242
+SYSTEMID.BBT             = 255
+
+
+# partition boot flag
+BOOTINDICATOR = v_types.venum()
+BOOTINDICATOR.NOBOOT = 0
+BOOTINDICATOR.SYSTEM_PARTITION = 128
+
+
+class PART_ENTRY(v_types.VStruct):
+    '''
+    partition entry in the MBR.
+    '''
+    def __init__(self):
+        super(PART_ENTRY, self).__init__()
```

Comment:
my weak technical explanation is the multiple-inheritance case, where i'd prefer Python to use its method resolution order to figure out the order to call things. i figure this leads to fewer corner cases or bugs down the road. but, since i don't use multiple inheritance too often, i don't have enough experience to really sell this style on the technical merits.

stylistically, though, i prefer using `super(ClassName, self)`. its more DRY. i find that having to repeat the super classes in the constructor is duplication of the class details, and i cant think of a reason why i *wouldn't* want to call the constructors (so this opens the possibility of adding a super class to the class declaration `class MyClass(ParentA, ParentB)`, but forgetting to update `__init__`).

i believe in python3, you can now do `super().__init__()` and things will *just work*, which i think is even more DRY, and more attractive.